code/modules/modular_armor/modular.dm

```diff
@@ -418,11 +418,15 @@
 
 /obj/item/clothing/head/modular/update_overlays()
 	. = ..()
-	update_clothing_icon()
+
+	if(overlays)
+		cut_overlays()
+
+	if(installed_module)
+		add_overlay(image(installed_module.icon, installed_module.item_state))
```

Comment:
```suggestion
	if(installed_module)
		. += image(installed_module.icon, installed_module.item_state)
```

`update_overlays` actually does this already in update_icons it calls this proc to update the overlays